File: caesar.c
//Encrypt string with Ceaser's encryption by the specified key.
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define LENALPHA 26

int main(int argc, string argv[])
{
    //Check only one argument has been entered
    if (argc != 2)
    {
        printf("You have entered %i arguments, please enter only one.\n",  argc - 1);
        return 1;
    }
    //Check argument is valid number
    for (int i = 0, n = strlen(argv[1]) ; i < n ; i++)
    {
        //Skip negative sign
        (i == 0) ? ((argv[1][0] == '-') ? i++ : i) : i;

        if (!isdigit(argv[1][i]))
        {
            printf("You have entered %s, please enter a number.\n",  argv[1]);
            return 1;
        }

    }


    //Turn Key into positive integer < length alphabet
    int key = atoi(argv[1]);
    key %= LENALPHA;
    key += LENALPHA;
    key %= LENALPHA;

    //Get users text to encrypt.
    string plaintext = get_string("plaintext: ");
    string ciphertext = plaintext;

    //Apply Ceaser's encryption to each character that is a letter
    char c;
    for (int i = 0, n = strlen(plaintext); i < n ; i++)
    {
        c = plaintext[i];

        //Check letter and case
        if isalpha(c)
        {
            char initial_letter = isupper(c) ? 'A' : 'a';

            //Apply Encryption
            c -= initial_letter;
            c += key;
            c %= LENALPHA;
            c += initial_letter;
        }

        ciphertext[i] = c;

    }

    //Return ciphertext to screen
    printf("ciphertext: %s\n", ciphertext);
    return 0;
}

File: caesar.py
"""Caesar Homework Problem"""
from cs50 import get_string


# Define global varaibles
LENALPHA = 26


def main(argv):
    """Encrypt plaintext message with key given using Ceasar's cipher."""

    # Ensure correct number of arguements
    if len(argv) != 2:

        # Raise program usage error
        raise Exception("Usage: python3 caesar.py key")

    # Ensure valid key
    try:
        # Retrieve key from input
        key = int(argv[1])

        # Raise Error if negative
        if key < 0:
            raise ValueError

    # If not an integer or negative
    except ValueError:

        # Raise key usage error
        raise Exception("Usage - key: must be a positive integer") from None

    # Retrieve key from input
    key = int(argv[1])

    # Retrieve plain text
    plain = get_string("plaintext: ")

    # Initiate ciphertext varaible
    cipher = ""

    # Iterate over characters in plain
    for char in plain:

        # Check if char is a capital
        if char.isupper():

            # Convert character to alphabetical index
            aIndex = ord(char) - ord('A')

            # Apply Caesar's Cipher
            aIndex = (aIndex + key) % LENALPHA

            # Convert alphabetical index back to character
            char = chr(aIndex + ord('A'))

        # Check if char is a capital
        elif char.islower():

            # Convert character to alphabetical index
            aIndex = ord(char) - ord('a')

            # Apply Caesar's Cipher
            aIndex = (aIndex + key) % LENALPHA

            # Convert alphabetical index back to character
            char = chr(aIndex + ord('a'))

        # Add char to ciphertext
        cipher += char

    # Print ciphertext to user
    print("ciphertext: " + cipher)


if __name__ == "__main__":
    # If program run get programs arguements
    from sys import argv

    # Call main with argv
    main(argv)
